Bluetooth: Host: Fix unable to create periodic adv sync

gangheivt opened the pull request:
In option of HCI_LE_Periodic_Advertising_Create_Sync, bit 2 depend on Periodic Advertising ADI Support feature.

Commit:
bluetooth: host: scan: Format source code

Run clang-format on scan.c

Signed-off-by: Gang He <ganghe@sifli.com>

Thalley commented on subsys/bluetooth/host/scan.c:
```diff
@@ -1970,8 +1970,10 @@ int bt_le_per_adv_sync_create(const struct bt_le_per_adv_sync_param *param,
+		if (BT_FEAT_LE_PER_ADV_ADI_SUPP(bt_dev.le.features)) {
+			cp->options |=
+				BT_HCI_LE_PER_ADV_CREATE_SYNC_FP_FILTER_DUPLICATE;
+		}
```
If this is the case, then I think we should reject the request early. 

At the top of this function we have a similar check

```c
	if (!BT_FEAT_LE_EXT_PER_ADV(bt_dev.le.features)) {
		return -ENOTSUP;
	}
```

We should not silently ignore parameters from the application, and we should perform parameter checks before attempting to allocate buffers and PA sync objects, so I suggest to move the check to above (but keep the assignment here)

cvinayak replied:
> We should not silently ignore parameters from the application

Agree to this. Application can check and use the correct parameters.

> This will make sample failed on chipset that do not support this feature.

Fix the sample to use the feature if supported and document it accordingly.

Commit:
samples: bluetooth: bap_broadcast_sink: format source code

Run clang-format to format sample source code.

Signed-off-by: Gang He <ganghe@sifli.com>

Thalley reviewed (changes requested):
The (actual) changes look overall good, with some minor suggestions/requests

cvinayak reviewed (dismissed):
Keep all the formatting changes in separate commit.

Commit:
bluetooth: host: Add parameter check when creating periodic adv sync

In option of HCI_LE_Periodic_Advertising_Create_Sync, bit 2
depend on Periodic Advertising ADI Support feature.

Signed-off-by: Gang He <ganghe@sifli.com>

Commit:
samples: bluetooth: bap_broadcast_sink: add parameter checking

BT_LE_PER_ADV_SYNC_OPT_FILTER_DUPLICATE option is only available when
BT_FEAT_LE_PER_ADV_ADI_SUPP is enabled in chipset.

Signed-off-by: Gang He <ganghe@sifli.com>

gangheivt commented on samples/bluetooth/bap_broadcast_sink/sample.yaml:
I added a separate commit to format main.c

cvinayak reviewed (commented):
LGTM. A nitpick, since the most line of files are formatted anyway, may be add `U` to unsigned assignments?

to `for (uint8_t i = 0U;`  etc.

gangheivt commented on samples/bluetooth/bap_broadcast_sink/src/main.c:
```diff
@@ -1025,9 +1022,21 @@ static int stop_adv(void)
+	if (BT_FEAT_LE_PER_ADV_ADI_SUPP(feature.features)) {
+		create_params.options = BT_LE_PER_ADV_SYNC_OPT_FILTER_DUPLICATE;
+	} else {
+		create_params.options = 0;
```
Fixed.

gangheivt commented on samples/bluetooth/bap_broadcast_sink/src/main.c:
```diff
@@ -1025,9 +1022,21 @@ static int stop_adv(void)
+	if (BT_FEAT_LE_PER_ADV_ADI_SUPP(feature.features)) {
+		create_params.options = BT_LE_PER_ADV_SYNC_OPT_FILTER_DUPLICATE;
+	} else {
+		create_params.options = 0U;
```
Updated.

Thalley reviewed (approved):
Thanks for the updates :) LGTM
